core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -591,22 +595,24 @@ private void seekInternalWithCachingPartiallyReadBlock(long pos) throws IOExcept
     // Precompute this because mPos will be updated several times in this function.
     final boolean isInCurrentBlock = pos / mBlockSize == mPos / mBlockSize;
 
-    // Make sure that mCurrentBlockInStream and mCurrentCacheStream is updated.
-    // mPos is not updated here.
-    updateStreams();
+    if (!(mPos == 0 && mCurrentBlockInStream == null && !isInCurrentBlock)) {
```

Comment:
I can negate the condition, but the idea is not to cache the first block if this is the first seek outside the first block.